ols/src/tools/oc_cli.py

```diff
@@ -0,0 +1,239 @@
+"""OpenShift CLI tools."""
+
+import logging
+import os
+import subprocess
+from typing import Optional
+
+from langchain.tools import tool
+
+logger = logging.getLogger(__name__)
+
+
+# TODO: OLS-1443
+# TODO: might also sanitize the case when llm sends the full command instead
+# of just args, eg. [oc get pods -n bla] -> [pods -n bla]
+def sanitize_oc_args(args: list[str]) -> list[str]:
+    """Sanitize `oc` CLI arguments."""
+    blocked_chars = {";", "&", "|", "`", "$", "(", ")", "<", ">", "\\"}
+
+    sanitized_args = []
+    for arg in args:
+        safe_part = []
+        for char in arg:
+            if char in blocked_chars:
+                # stop processing further characters in this argument
+                logger.warning(
+                    "Problematic character(s) found in oc tool argument '%s'", arg
+                )
+                break
+            safe_part.append(char)
+
+        if safe_part:  # only add non-empty results
+            sanitized_args.append("".join(safe_part).strip())
+
+    return sanitized_args
+
+
+def run_oc(args: list[str]) -> subprocess.CompletedProcess:
+    """Run `oc` CLI with provided arguments and command."""
+    try:
+        res = subprocess.run(  # noqa: S603
```

Comment:
@onmete We are not passing the token to the actual command that is suggested by the model - We need to execute the command with the user token . Example : `oc describe pod lightspeed-operator-controller-manager-bc4576dbc-wfbh8  -n default --token="e..."` -  Can be done with security story

Reply:
For each user question, the token is checked and used to login. All tool calls within the user question are executed with this.
Yes, I see the problem that the service is logged to oc until another question hits.
We can spawn a dedicated shell for the user question execution, that will go away with the response.
Or as you say, sending the token and server for each command.
Related to the comment bellow - I guess there is technically nothing wrong if the tools fails because of wrong token, just in this specific case we can know in advance. But we can go with failures for now and solve it later.